src/Analyzer.Utilities/DiagnosticHelpers.cs

```diff
@@ -98,5 +99,29 @@ public static string GetMemberName(ISymbol symbol)
             // For other language constructs
             return symbol.Name;
         }
+
+        public static string GetMeaningfulParentName(SyntaxNode current, SemanticModel model)
```

Comment:
I replied this in somewhere to Sri before. The current GetEnclosingSymbol will return empty string for code like `Task.Run(()=> { ...}), which makes the diagnostic message looks weird. 

This method will return non-empty enclosing method or class's name instead.